crates/rolldown_ecmascript_utils/src/ast_snippet.rs

```diff
@@ -62,47 +62,40 @@ impl<'ast> AstSnippet<'ast> {
     object: ast::Expression<'ast>,
     names: &[CompactStr],
     span: Span,
+    mut cursor: usize,
   ) -> ast::Expression<'ast> {
-    match names {
-      [] => object,
-      _ => {
-        let prop_name = names[names.len() - 1].as_str();
-        let object = self.member_expr_or_ident_ref(object, &names[0..names.len() - 1], span);
-        if identifier::is_identifier_name(prop_name) {
-          ast::Expression::StaticMemberExpression(self.builder.alloc_static_member_expression(
-            span,
-            object,
-            self.id_name(prop_name, span),
-            false,
-          ))
-        } else {
-          ast::Expression::ComputedMemberExpression(self.builder.alloc_computed_member_expression(
-            span,
-            object,
-            self.builder.expression_string_literal(span, self.builder.atom(prop_name), None),
-            false,
-          ))
-        }
-      }
+    let mut cur = object;
+    while cursor < names.len() {
+      let name = names[cursor].as_str();
```

Comment:
With the change above, I guess we can change this into a for-loop (which may reduce the boundary check).
```suggestion
    for name in names {
      let name = name.as_str();
```